website/src/hooks/tasks/useGenericTaskAPI.tsx

```diff
@@ -1,26 +1,43 @@
 import { useState } from "react";
 import { get, post } from "src/lib/api";
-import { BaseTask, TaskResponse, TaskType as TaskTypeEnum } from "src/types/Task";
+import { TaskApiHook } from "src/types/Hooks";
+import { BaseTask, TaskAvailableResponse, TaskResponse, TaskType as TaskTypeEnum } from "src/types/Task";
 import useSWRImmutable from "swr/immutable";
 import useSWRMutation from "swr/mutation";
 
-export const useGenericTaskAPI = <TaskType extends BaseTask>(taskType: TaskTypeEnum) => {
-  type ConcreteTaskResponse = TaskResponse<TaskType>;
+export const useGenericTaskAPI = <TaskType extends BaseTask>(taskType: TaskTypeEnum): TaskApiHook<TaskType> => {
+  const [response, setReponse] = useState<TaskResponse<TaskType>>({ status: "AWAITING_INITIAL" });
+  const [isLoading, setIsLoading] = useState(true);
```

Comment:
I am not sure how  the behavior of `isLoading` in this hook is different from the default one from  `useSWR`.

Reply:
The difference, which I only just found out about (I was all ready to change to isLoading from useSWR), is that the isLoading for useSWRImmutable doesn't change back to false when you call mutate().

Reply:
>  isLoading for useSWRImmutable doesn't change back to false when you call mutate().

That is a good point, could you please write it as a comment in the code

Reply:
Good idea